In [1]:
#import required modules
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("https://raw.githubusercontent.com/YBI-Foundation/Dataset/main/Credit%20Default.csv",sep=",")

In [3]:
df

Income        Age         Loan  Loan to Income  Default
0     66155.92510  59.017015  8106.532131        0.122537        0
1     34415.15397  48.117153  6564.745018        0.190752        0
2     57317.17006  63.108049  8020.953296        0.139940        0
3     42709.53420  45.751972  6103.642260        0.142911        0
4     66952.68885  18.584336  8770.099235        0.130990        1
...           ...        ...          ...             ...      ...
1995  59221.04487  48.518179  1926.729397        0.032535        0
1996  69516.12757  23.162104  3503.176156        0.050394        0
1997  44311.44926  28.017167  5522.786693        0.124636        1
1998  43756.05660  63.971796  1622.722598        0.037086        0
1999  69436.57955  56.152617  7378.833599        0.106267        0

[2000 rows x 5 columns]

In [4]:
df.head()

Income        Age         Loan  Loan to Income  Default
0  66155.92510  59.017015  8106.532131        0.122537        0
1  34415.15397  48.117153  6564.745018        0.190752        0
2  57317.17006  63.108049  8020.953296        0.139940        0
3  42709.53420  45.751972  6103.642260        0.142911        0
4  66952.68885  18.584336  8770.099235        0.130990        1

In [5]:
df.tail()

Income        Age         Loan  Loan to Income  Default
1995  59221.04487  48.518179  1926.729397        0.032535        0
1996  69516.12757  23.162104  3503.176156        0.050394        0
1997  44311.44926  28.017167  5522.786693        0.124636        1
1998  43756.05660  63.971796  1622.722598        0.037086        0
1999  69436.57955  56.152617  7378.833599        0.106267        0

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Income          2000 non-null   float64
 1   Age             2000 non-null   float64
 2   Loan            2000 non-null   float64
 3   Loan to Income  2000 non-null   float64
 4   Default         2000 non-null   int64  
dtypes: float64(4), int64(1)
memory usage: 78.2 KB


In [7]:
df.columns.tolist()

['Income', 'Age', 'Loan', 'Loan to Income', 'Default']

In [8]:
df.describe()

Income          Age          Loan  Loan to Income      Default
count   2000.000000  2000.000000   2000.000000     2000.000000  2000.000000
mean   45331.600018    40.927143   4444.369695        0.098403     0.141500
std    14326.327119    13.262450   3045.410024        0.057620     0.348624
min    20014.489470    18.055189      1.377630        0.000049     0.000000
25%    32796.459720    29.062492   1939.708847        0.047903     0.000000
50%    45789.117310    41.382673   3974.719418        0.099437     0.000000
75%    57791.281670    52.596993   6432.410625        0.147585     0.000000
max    69995.685580    63.971796  13766.051240        0.199938     1.000000

In [9]:
df.describe(include='all')

Income          Age          Loan  Loan to Income      Default
count   2000.000000  2000.000000   2000.000000     2000.000000  2000.000000
mean   45331.600018    40.927143   4444.369695        0.098403     0.141500
std    14326.327119    13.262450   3045.410024        0.057620     0.348624
min    20014.489470    18.055189      1.377630        0.000049     0.000000
25%    32796.459720    29.062492   1939.708847        0.047903     0.000000
50%    45789.117310    41.382673   3974.719418        0.099437     0.000000
75%    57791.281670    52.596993   6432.410625        0.147585     0.000000
max    69995.685580    63.971796  13766.051240        0.199938     1.000000

In [10]:
df['Default'].value_counts()

0    1717
1     283
Name: Default, dtype: int64

In [12]:
#Let's split the data
Y = df['Default']

In [13]:
X = df[['Income', 'Age', 'Loan', 'Loan to Income']]

In [15]:
Y.shape

(2000,)

In [16]:
X.shape

(2000, 4)

In [17]:
#as the prediction column is a classification model

In [18]:
from sklearn.model_selection import train_test_split

In [19]:
X_train,X_test,Y_train,Y_test = train_test_split(X,Y,random_state=42)

In [20]:
X_train.shape

(1500, 4)

In [21]:
X_test.shape

(500, 4)

In [22]:
Y_train.shape

(1500,)

In [23]:
Y_test.shape

(500,)

In [24]:
#model Building

In [25]:
from sklearn.linear_model import LogisticRegression

In [26]:
LR = LogisticRegression()

In [27]:
LR.fit(X_train,Y_train)

LogisticRegression()

In [28]:
LR.intercept_

array([9.53580211])

In [29]:
LR.coef_

array([[-2.34494425e-04, -3.53529255e-01,  1.72098433e-03,
         1.54930422e+00]])

In [30]:
y_pred = LR.predict(X_test)

In [31]:
y_pred

array([1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1,
       0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 0,
       0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1,

In [32]:
#let's see the accuracy metrics

In [33]:
from sklearn.metrics import classification_report,confusion_matrix,accuracy_score

In [35]:
print(classification_report(Y_test,y_pred))

              precision    recall  f1-score   support

           0       0.96      0.97      0.97       422
           1       0.82      0.81      0.81        78

    accuracy                           0.94       500
   macro avg       0.89      0.89      0.89       500
weighted avg       0.94      0.94      0.94       500



In [36]:
print(confusion_matrix(Y_test,y_pred))

[[408  14]
 [ 15  63]]


In [37]:
print(accuracy_score(Y_test,y_pred))

0.942


# **Therefore we can say that our model has an accuracy of 94%**